bee-test/src/rand/metadata.rs

```diff
@@ -13,7 +13,7 @@ use bee_tangle::{
 
 /// Generates a random conflict reason.
 pub fn rand_conflict_reason() -> ConflictReason {
-    ((rand_number::<u64>() % 7) as u8).try_into().unwrap()
```

Comment:
Are you intentionally leaving out `ConflictReason::InvalidDustAllowance` and ``ConflictReason::SemanticValidationFailed`? If so, we should maybe document that.

Ideally we would have something like this though:

```rust
let mut rng = rand::thread_rng();
let conflicts = [
    ConflictReason::None,
    ConflictReason::InputUtxoAlreadySpent,
    ...
];
let conflict = conflicts.choose(&mut rng).unwrap();
```

With this we could roundtrip: From _random conflict_ to `u8` and then `try_into`.

Reply:
`ConflictReason::InvalidDustAllowance` has been removed and yeah we're leaving out `ConflictReason::SemanticValidationFailed` intentionally.  I will leave a comment for now, this PR is really just the bare minimum to make every comment, the `bee-test` will be drastically refactored in an incoming PR.